bigquery-storage/api/BigQueryStorage.Samples/AppendRowsPending.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START bigquerystorage_append_rows_pending]
+
+using Google.Api.Gax.Grpc;
+using Google.Cloud.BigQuery.Storage.V1;
+using Google.Protobuf;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using static Google.Cloud.BigQuery.Storage.V1.AppendRowsRequest.Types;
+
+public class AppendRowsPendingSample
+{
+    /// <summary>
+    /// This code sample demonstrates how to write records in pending mode.
+    /// Create a write stream, write some sample data, and commit the stream to append the rows.
+    /// The CustomerRecord proto used in the sample can be seen in Resources folder and generated C# is placed in Data folder in the Samples directory.
+    /// </summary>
+    public async Task AppendRowsPendingAsync(string projectId, string datasetName, string tableName)
+    {
+        BigQueryWriteClient bigQueryWriteClient = await BigQueryWriteClient.CreateAsync();
+        // Initialize a write stream for the specified table.
+        // When creating the stream, choose the type. Use the PENDING type to wait
+        // until the stream is committed before it is visible. See:
+        // https://cloud.google.com/bigquery/docs/reference/storage/rpc/google.cloud.bigquery.storage.v1#google.cloud.bigquery.storage.v1.WriteStream.Type
+        WriteStream stream = new WriteStream { Type = WriteStream.Types.Type.Pending };
+        TableName table = TableName.FromProjectDatasetTable(projectId, datasetName, tableName);
+
+        // This header is required, if dataset is located in non-default region, so that the BigQuery Storage API knows which region to route the request to.
+        CallSettings callSettings = CallSettings.FromHeader("x-goog-request-params", $"writeStream={stream.Name}");
+        stream = await bigQueryWriteClient.CreateWriteStreamAsync(table, stream, callSettings);
```

Comment:
I wouldn't expect you to need the callSettings for this call - I'd expect it to be in the AppendRows call.
I'd also expect it to be `write_stream` rather than `writeStream`.

Reply:
Passed the callSettings to AppendRows call (My bad...I checked the stream variable in nodeJS sample and passed it in wrong stream).  Changed writeStream to write_stream. (Python uses write_stream, while nodeJS uses writeStream). 
My tests are passing with and without header in EU as well as Tokyo.